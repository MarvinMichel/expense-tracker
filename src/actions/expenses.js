import { v1 as uuid } from 'uuid';
import database from '../firebase/firebase';

export const addExpense = expense => ({
  type: 'ADD_EXPENSE',
  expense
});

export const startAddExpense = (expenseData ={}) => {
  return (dispatch) => {
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData;
    const expense = { description, note, amount, createdAt };
    
    return database.ref('expenses').push(expense).then(ref => {
      dispatch(addExpense({
        id: ref.key,
        ...expense
      }));
    })
  };
};

export const removeExpense = ({ id } = {}) => ({
  type: 'REMOVE_EXPENSE',
  id,
});

export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id,
  updates,
});
