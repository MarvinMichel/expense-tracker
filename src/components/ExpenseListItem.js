import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import 'numeral/locales/nl-nl';

numeral.locale('nl-nl');

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <li className="list-item">
    <Link to={ `/edit/${id}` }>
      <div>
        <h3 className="list-item__title">{ description }</h3>
        <span className="list-item__sub-title">{ moment(createdAt).format('DD MMMM YYYY') }</span>
      </div>
      <h3 className="list-item__data">{ numeral(amount / 100).format('$0,0.00') }</h3>
    </Link>
  </li>
);

export default ExpenseListItem;
