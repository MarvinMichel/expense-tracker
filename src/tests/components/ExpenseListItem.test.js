import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render expense list item', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
