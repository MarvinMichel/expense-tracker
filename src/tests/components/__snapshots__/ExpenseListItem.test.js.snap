// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    195
     - 
    0
  </p>
</div>
`;
